### ISSUE TITLE
CA1001 triggered for IAsyncDisposable classes

### ISSUE THREAD

**Issue by colinthompson** [dotnet/roslyn-analyzers#3371]
I'm receiving a code analysis warning for CA1001 for types implementing `IAsyncDisposable`.

It looks like this was raised in #2325 and fixed in #2916, but the fix hasn't been included in a release yet.

#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

[CA1001](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca1001?view=vs-2019)

#### Repro steps

```
namespace Example
{
    public class Test : IAsyncDisposable
    {
        private DbConnection connection = new SqliteConnection("Data Source=:memory:");

        public ValueTask DisposeAsync() => connection.DisposeAsync();
    }
}
```

#### Expected behavior

No code analysis warning for CA1001.

#### Actual behavior

Code analysis warning for CA1001.

**Comment by Evangelink** [comment]
@mavasani Are you planning to have a new release soon? I confirm the ticket is handled but not yet released.

**Comment by mavasani** [comment]
@Evangelink Yes, publishing a new beta early this week is on my plate.

**Comment by giggio** [comment]
Hello there,
Is this released yet?

**Comment by AdamWillden** [comment]
The commits from  #2916 are tagged as being in 3.0.0

I just asked a colleague to remove a suppression now that we've updated to 3.0.0 and can confirm the suppression can be removed. 

Time to close this?

**Comment by Evangelink** [comment]
Ping @mavasani

**Comment by giggio** [comment]
This is not fixed yet.

![image](https://user-images.githubusercontent.com/334958/80555961-56ae5c80-89a8-11ea-8d9c-933ecb8f74b6.png)

Test case:

````csharp
public class Test : IAsyncDisposable
{
    private MemoryStream ms = new MemoryStream();

    public ValueTask DisposeAsync() => ms.DisposeAsync();
}
````

.NET Core version 3.1.201 (latest released at the moment).

Error also shows up on Microsoft Visual Studio Enterprise 2019 Version 16.5.4.

**Comment by Evangelink** [comment]
@giggio Could you let us know the version of the analyzer package you are using?

**Comment by giggio** [comment]
I was on 2.9.8, so I went on to check, there was a release (3.0.0) 2 days ago. After updating, the warning is gone. Thanks.
